lib/ethon/easy/response_callbacks.rb

```diff
@@ -39,6 +39,7 @@ def on_headers(&block)
       # @example Execute on_headers.
       #   request.headers
       def headers
+        @headers_called = true
```

Comment:
could you check `@headers_called` in here? That way you can call it from the outside without the check.